src/services/doppler-billing-api-client.ts

```diff
@@ -0,0 +1,106 @@
+import { ResultWithoutExpectedErrors } from '../doppler-types';
+import { AxiosInstance, AxiosStatic } from 'axios';
+import { AppSession } from './app-session';
+import { RefObject } from 'react';
+
+export interface DopplerBillingApiClient {
+  getInvoices(page: number, pageSize: number): Promise<ResultWithoutExpectedErrors<Invoices>>;
+}
+
+interface DopplerBillingApiConnectionData {
+  jwtToken: string;
+  idUser: string;
+}
+
+export interface Invoices {
+  items: Invoice[];
+  totalItems: number;
+}
+
+export interface Invoice {
+  accountId: string;
+  product: string;
+  date: string;
+  currency: string;
+  amount: number;
+  filename: string;
+}
+
+export class HttpDopplerBillingApiClient implements DopplerBillingApiClient {
+  private readonly axios: AxiosInstance;
+  private readonly baseUrl: string;
+  private readonly connectionDataRef: RefObject<AppSession>;
+
+  constructor({
+    axiosStatic,
+    baseUrl,
+    connectionDataRef,
+  }: {
+    axiosStatic: AxiosStatic;
+    baseUrl: string;
+    connectionDataRef: RefObject<AppSession>;
+  }) {
+    this.baseUrl = baseUrl;
+    this.axios = axiosStatic.create({
+      baseURL: this.baseUrl,
+    });
+    this.connectionDataRef = connectionDataRef;
+  }
+
+  private getDopplerBillingApiConnectionData(): DopplerBillingApiConnectionData {
+    const connectionData = this.connectionDataRef.current;
+    if (
+      !connectionData ||
+      connectionData.status !== 'authenticated' ||
+      !connectionData.jwtToken ||
+      !connectionData.userData
+    ) {
+      throw new Error('Doppler Billing API connection data is not available');
+    }
+    return {
+      jwtToken: connectionData.jwtToken,
+      idUser: connectionData ? connectionData.userData.user.idUser : '',
+    };
+  }
+
+  private mapInvoices(data: any): Invoice[] {
+    return data.map((x: any) => ({
+      accountId: x.accountId,
+      product: x.product,
+      date: x.date.toString('dd/mm/yyyy'),
```

Comment:
Please, always use plain dates (or moment objects), but never formatted values, the output of our clients should be flexible data independent of visualization.

Reply:
Yes, good. I will update it.